### PR TITLE
[ready] Craftable Prescription & Sunglasses HUDs

### LINKED ISSUE #11286: Prescription Not Available
Issue reported from  Round ID: 21762 (Citadel Station 13)

Reporting client version: 513

Combining a regular Prescription HUD of either medical or security kind with sunglasses to make a Sunglasses HUD removes the prescription, leaving you without usable eyewear.

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request
closes #11286
No longer can use Prescription Glasses/Huds for crafting
New glasses that are both HUDs and prescriptions!
New Diagnostic Sunglasses!
New crafting for each set of Sunglasses
Ect ect
## Why It's Good For The Game

Consistency and such as their is no way to get Prescription sunglasses HUD's
People dislike not having to chose from cool shades from being able to see!

## Changelog
:cl:
balance: Added sunglasses that are able to be huds AND prescription 
add: Prescription sunglasses and crafting of each new type for - Diagnostic, Med and Sec
add: Diagnostic Sunglasses
fix: Blackists Prescription HUDs from sunglasses crafting
/:cl:

### CODE REVIEW

**Comment by Ragolution** [comment]
Might be a heavy-handed solution for a simple (or non-) problem. Depends on what the response for #11286 is.

**Comment by YakumoChen** [comment]
I mean, "flash protection" versus "being able to see" is sort of the point of a tradeoff.
People don't like being able to choose, sure, but _obviously_. Maybe they shouldn't have fucked up eyes in the first place?

**Comment by Ragolution** [comment]
~~@YakumoChen 
They still have to choose with this PR. You don't get both, this just allows you to craft prescription HUDs and sunglasses HUDs with a respective HUD and a pair of the respective glasses.~~
Wait I'm wrong. LILY NO THIS ISNT WHAT WE DISCUSSED.

~~@Trilbyspaceclone You forgot mesons. Probably rename this "Craftable Prescription & Sunglasses HUDs".~~

**Review by Ghommie — CHANGES_REQUESTED** [review]
A more reliable long term solution to the issue, without bloating the crafting menu:
```
/obj/item/clothing/glasses/hud/CheckParts(list/parts_list)
	. = ..()
	if(vision_correction)
		return
	for(var/obj/item/clothing/glasses/G in parts_list)
		if(G.vision_correction)
			vision_correction = TRUE
			name = "prescription [name]"
			return
```
Undo your changes.

**Comment by Ragolution** [comment]
> A more reliable long term solution to the issue, without bloating the crafting menu:
> 
> ```
> /obj/item/clothing/glasses/hud/CheckParts(list/parts_list)
> 	. = ..()
> 	if(vision_correction)
> 		return
> 	for(var/obj/item/clothing/glasses/G in parts_list)
> 		if(G.vision_correction)
> 			vision_correction = TRUE
> 			name = "prescription [name]"
> 			return
> ```
> 
> Undo your changes.

I'm currently writing what I would consider an 'alternative' PR for this issue. What does this do, exactly? From my vantage it appears to just rename the HUDs with the vision correction flag as prescription. Does that interact as-desired with crafting?

Edit: I think I made sense of it. Seems to have a lot more finesse.

File: code/datums/components/crafting/recipes/recipes_clothing.dm [hunk 1]
```diff
@@ -130,7 +156,6 @@
 	reqs = list(/obj/item/stack/sheet/durathread = 7,
 				/obj/item/stack/sheet/leather = 3)
 	time = 70
-	always_availible = TRUE
 	category = CAT_CLOTHING
```
Comment (Ragolution) [inline code/datums/components/crafting/recipes/recipes_clothing.dm:162]: Broken commenting on Stunglasses. Fix this.

File: code/modules/clothing/glasses/hud.dm [hunk 2]
```diff
@@ -80,6 +116,11 @@
 	vision_correction = 1
 	glass_colour_type = /datum/client_colour/glass_colour/lightorange
 
+/obj/item/clothing/glasses/hud/diagnostic/sunglasses/prescription
```
Comment (Ragolution) [inline code/modules/clothing/glasses/hud.dm:113]: you defined diagnostic prescription sunglasses twice in the same file.

File: code/modules/clothing/glasses/hud.dm [hunk 3]
```diff
@@ -65,13 +79,35 @@
...
+/obj/item/clothing/glasses/hud/diagnostic/sunglasses
+	name = "diagnostic HUDSunglasses"
+	desc = "Sunglasses with a diagnostic HUD."
+	icon_state = "sunhuddiagnostic"
+	darkness_view = 1
+	flash_protect = 1
+	tint = 1
+
+/obj/item/clothing/glasses/hud/diagnostic/sunglasses/prescription
+	name = "diagnostic HUDSunglasses"
```
Comment (Ragolution) [inline code/modules/clothing/glasses/hud.dm]: improper naming

**Review by Ragolution — CHANGES_REQUESTED** [review]
There's a bunch of weird grammar and you defined diagnostic prescription sunglasses twice in the same file.

This also needs the Balance tag since you've added prescription sunglasses.

File: code/datums/components/crafting/recipes/recipes_clothing.dm (hunk 1 above)
Comment (Trilbyspaceclone) [reply code/datums/components/crafting/recipes/recipes_clothing.dm:162]: ?

Comment (Ragolution) [reply code/datums/components/crafting/recipes/recipes_clothing.dm:162]: ~~You broke where Stunglasses were commented out.~~

Git hub made it look like it was only half-commented. smh.

File: code/datums/components/crafting/recipes/recipes_clothing.dm [hunk 4]
```diff
@@ -56,9 +61,10 @@
 	result = /obj/item/clothing/glasses/hud/health/sunglasses
 	time = 20
 	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
+	parts = list(/obj/item/clothing/glasses/hud/health = 1)
```
Comment (Ghommie) [inline code/datums/components/crafting/recipes/recipes_clothing.dm]:
```suggestion
	parts = list(/obj/item/clothing/glasses/hud/health = 1, 
					/obj/item/clothing/glasses/sunglasses = 1)
```

File: code/datums/components/crafting/recipes/recipes_clothing.dm [hunk 5]
```diff
@@ -86,6 +92,26 @@
 	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
 	reqs = list(/obj/item/clothing/glasses/sunglasses/reagent = 1)
 	category = CAT_CLOTHING
+
+/datum/crafting_recipe/diagnostic_sunglasses
+	name = "Diagnostic HUDsunglasses"
+	result = /obj/item/clothing/glasses/hud/diagnostic/sunglasses
+	time = 20
+	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
+	parts = list(/obj/item/clothing/glasses/hud/diagnostic = 1)
```
Comment (Ghommie) [inline code/datums/components/crafting/recipes/recipes_clothing.dm]:
```suggestion
	parts = list(/obj/item/clothing/glasses/hud/diagnostic = 1,
					/obj/item/clothing/glasses/sunglasses = 1)
```

File: code/modules/clothing/glasses/hud.dm [hunk 6]
```diff
@@ -65,13 +79,30 @@
 	tint = 1
 	glass_colour_type = /datum/client_colour/glass_colour/blue
 
+/obj/item/clothing/glasses/hud/health/sunglasses/prescription
+	name = "medical prescription HUDSunglasses"
+	desc = "Sunglasses with a medical HUD and a second layer of normal prescription glasses."
```
Comment (Ghommie) [inline code/modules/clothing/glasses/hud.dm]:
```suggestion
	name = "prescription medical HUDSunglasses"
```

File: code/modules/clothing/glasses/hud.dm [hunk 7]
```diff
@@ -80,6 +111,11 @@
 	vision_correction = 1
 	glass_colour_type = /datum/client_colour/glass_colour/lightorange
 
+/obj/item/clothing/glasses/hud/diagnostic/sunglasses/prescription
+	name = "diagnostic prescription HUDSunglasses"
+	desc = "Sunglasses with a diagnostic HUD and a second layer of normal prescription glasses."
```
Comment (Ghommie) [inline code/modules/clothing/glasses/hud.dm]:
```suggestion
	name = "prescription diagnostic HUDSunglasses"
```

File: code/modules/clothing/glasses/hud.dm [hunk 8]
```diff
@@ -142,6 +181,11 @@
 	tint = 1
 	glass_colour_type = /datum/client_colour/glass_colour/darkred
 
+/obj/item/clothing/glasses/hud/security/sunglasses/prescription
+	name = "security HUDSunglasses"
+	desc = "Sunglasses with a security HUD and a second layer of normal prescription glasses.."
```
Comment (Ghommie) [inline code/modules/clothing/glasses/hud.dm]:
```suggestion
	name = "prescription security HUDSunglasses"
```

File: code/modules/clothing/glasses/hud.dm [hunk 9]
```diff
@@ -142,6 +181,11 @@
...
+	vision_correction = 1
```
Comment (Ghommie) [inline code/modules/clothing/glasses/hud.dm:184]:
```suggestion
	vision_correction = TRUE
```

File: code/modules/clothing/glasses/hud.dm [hunk 10]
```diff
@@ -80,6 +111,11 @@
 	vision_correction = 1
 	glass_colour_type = /datum/client_colour/glass_colour/lightorange
 
+/obj/item/clothing/glasses/hud/diagnostic/sunglasses/prescription
+	name = "diagnostic prescription HUDSunglasses"
+	desc = "Sunglasses with a diagnostic HUD and a second layer of normal prescription glasses."
+	vision_correction = 1
```
Comment (Ghommie) [inline code/modules/clothing/glasses/hud.dm:115]:
```suggestion
	vision_correction = TRUE
```

File: code/datums/components/crafting/recipes/recipes_clothing.dm [hunk 11]
```diff
@@ -33,11 +33,16 @@
...
+////////
+//Huds//
+////////
+
 /datum/crafting_recipe/hudsunsec
 	name = "Security HUDsunglasses"
 	result = /obj/item/clothing/glasses/hud/security/sunglasses
 	time = 20
 	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
+	parts = list(/obj/item/clothing/glasses/hud/security = 1)
```
Comment (Ghommie) [inline code/datums/components/crafting/recipes/recipes_clothing.dm]:
```suggestion
	parts = list(/obj/item/clothing/glasses/hud/security = 1,
					/obj/item/clothing/glasses/sunglasses = 1)
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Review by Ghommie — APPROVED** [review]

**Pull request merged**